drivers/adc/adc_mcux_lpadc.c

```diff
@@ -57,7 +59,107 @@ struct mcux_lpadc_data {
 	lpadc_conv_command_config_t cmd_config[CONFIG_LPADC_CHANNEL_COUNT];
 };
 
+static int mcux_lpadc_acquisition_time_setup(const struct device *dev, uint16_t acq_time,
+					     lpadc_conv_command_config_t *cmd)
+{
+	const struct mcux_lpadc_config *config = dev->config;
+	uint32_t adc_freq_hz = 0;
+	float adc_cycle_ns = 0.0;
+	/* The unit of elements in the sample time array is ADCK. */
+	float sample_time_array[] = {3.0, 5.0, 7.0, 11.0, 19.0, 35.0, 67.0, 131.0};
```

Comment:
why is it needed to use floating points here ?

Reply:
In this patch logic, I convert the sample time array elements unit from ADCK to ns, that is each element needs to be multiplied by the variable  **adc_cycle_ns**, and the variable adc_cycle_ns is float point type.